packages/transaction-controller/src/utils.ts

```diff
@@ -65,6 +70,12 @@ export function validateTxParams(txParams: TransactionParams) {
     );
   }
 
+  if (isEIP1559Transaction(txParams) && !isEIP1559Compatible) {
+    throw ethErrors.rpc.invalidParams(
```

Comment:
I've already included the usage of `eth-rpc-errors` package as probably the [PR](https://github.com/MetaMask/core/pull/1690) to use it will be merged first and I don't want to forget to use the standardized error here.